src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -1107,7 +1114,7 @@ private void initiatorCompleteOrWait()
     public synchronized void sessionFailed()
     {
         logger.error("[Stream #{}] Remote peer {} failed stream session.", planId(), peer.toString());
-        closeSession(State.FAILED);
+        closeSession(State.FAILED," Remote peer " + peer + " failed stream session");
```

Comment:
nit: formatting
```suggestion
        closeSession(State.FAILED, "Remote peer " + peer + " failed stream session");
```

Looks like there's some inconsistent formatting of the whitespace at the start of `failureReasons` - it's probably better to have the string builder add it between peers as it may be forgotten when adding new messages.